renderdoc/driver/ihv/nv/nv_counter_enumerator.cpp

```diff
@@ -126,6 +135,72 @@ void NVCounterEnumerator::Impl::InitEnumerateCounters()
 
   m_EnumerationDone = true;
 
+  if(!CounterAvailabilityImage.empty())
+  {
+    NVPA_Status result;
+    NVPW_RawCounterConfig_SetCounterAvailability_Params params = {};
+    params.structSize = sizeof(params);
+    params.pRawCounterConfig = RawCounterConfigBuilder;
+    params.pCounterAvailabilityImage = CounterAvailabilityImage.data();
+    result = NVPW_RawCounterConfig_SetCounterAvailability(&params);
+    if(result != NVPA_STATUS_SUCCESS)
+    {
+      NV_PERF_LOG_ERR(50, "NvPerf could not determine counter availability for this GPU");
+      return;
+    }
+  }
+
+  rdcarray<uint32_t> availableDomains;
+  {
+    const std::vector<NVPW_RawCounterDomain> availableSCDs =
+        RawCounterConfigBuilder.GetAllAvailableSingularCounterDomains();
+    for(NVPW_RawCounterDomain scd : availableSCDs)
+    {
+      availableDomains.push_back(static_cast<uint32_t>(scd));
```

Comment:
Please use normal casts rather than C++ casts.

```suggestion
      availableDomains.push_back((uint32_t)scd);
```